ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/ForkChoiceUtil.java

```diff
@@ -335,27 +338,46 @@ public static BlockImportResult on_block(
     // Update finalized checkpoint
     final Checkpoint finalizedCheckpoint = state.getFinalized_checkpoint();
     if (finalizedCheckpoint.getEpoch().compareTo(store.getFinalizedCheckpoint().getEpoch()) > 0) {
+      // TODO: Fork Choice Spec does not necessarily ask us to store this checkpoint state, so if we
+      // do error here and exit, we will possibly have acted differently than other nodes.
```

Comment:
The only case where this actually fails is if we're asking `process_slots` to go backwards (ie get to a slot prior to the starting state slot) - since we're definitely not doing that, the exception handling is just annoying boilerplate.

Per the spec we should be applying all of `on_attestation` to all attestations in the block which would trigger this same processing anyway.

Reply:
I see.